src/Infrastructure/Id/Uuid/UuidFactory.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bookkeeper\Accounting\Infrastructure\Id\Uuid;
+
+use InvalidArgumentException;
+use Ramsey\Uuid\Exception\UnableToBuildUuidException;
+use Ramsey\Uuid\UuidFactoryInterface;
+use Ramsey\Uuid\UuidInterface;
+
+final class UuidFactory
+{
+    public const MODE_STRING = 0x1;
+    public const MODE_BYTES = 0x2;
+    public const MODE_INTEGER = 0x4;
+    public const MODE_ALL = 0x7;
+
+    private UuidFactoryInterface $uuidFactory;
+    private int $generatedVersion;
+    private int $mode;
+
+    public function __construct(UuidFactoryInterface $uuidFactory, int $generateVersion = 4, int $mode = self::MODE_ALL)
```

Comment:
What use-case for $mode ?

Reply:
To prevent parsing strings in none-supported format to save some CPU time. E.g. when user knows he stores uuid in the int(128) format, he can set `mode` to the `UuidFactory::MODE_INTEGER` value. Or for the migration period it may be smthng like `UuidFactory::MODE_INTEGER | UuidFactory::MODE_STRING`.

Reply:
See `UuidFactory::parse()` to get the rest of this idea.

Reply:
ID often show to user and used in all layers of application. This is will be difficult when value is bytes or integer type. I think better always use string mode and typecast for save to DB only in repository.